backend/tracim_backend/app_models/email_validators.py

```diff
@@ -0,0 +1,57 @@
+import re
+
+from marshmallow import ValidationError
+from marshmallow.validate import Validator
+
+
+class TracimEmailValidator(Validator):
+    """
+    Special Email Validator who accept email from subdomain
+    - check existing "@".
+    - check if both part are not empty.
+    - refuse special characters that cause issue with rfc style address parsing used
+    in frontend and backend like "<" and ";"
+    """
+
+    USER_REGEX = re.compile(r"^[^<>,;\n]+$")
+    DOMAIN_REGEX = re.compile(r"^[^<>,;\n]+$")
+
+    default_message = "Not a valid email address."
+
+    def __init__(self, error=None):
+        self.error = error or self.default_message
+
+    def _format_error(self, value):
+        return self.error.format(input=value)
+
+    def __call__(self, value):
+        message = self._format_error(value)
+
+        if not value or "@" not in value:
+            raise ValidationError(message)
+
+        user_part, domain_part = value.rsplit("@", 1)
+
+        if not self.USER_REGEX.match(user_part):
+            raise ValidationError(message)
+        if self.DOMAIN_REGEX.match(domain_part):
+            return value
+        raise ValidationError(message)
```

Comment:
```suggestion
    FORBIDDEN_CHARS = re.compile(r"[<>,;\n]")

    default_message = "Not a valid email address."

    def __init__(self, error=None):
        self.error = error or self.default_message

    def _format_error(self, value):
        return self.error.format(input=value)

    def __call__(self, value):
        message = self._format_error(value)

        if not value or "@" not in value or self.FORBIDDEN_CHARS.match(user_part):
            raise ValidationError(message)

        user_part, domain_part = value.rsplit("@", 1)

        if not user_part or not domain_part:
            raise ValidationError(message)
```
(not tested, solves duplicate regex, make it clear that we forbid empty parts)

Reply:
edited to fix some mistakes

Reply:
This does not work "a@>" is valid and I don't like the matching in inverted side, it's really confusing me.

Reply:
i made a better version of my code, removing duplicate regex, and reordering  the last part, but keeping regex for validation of full part.

Reply:
Ah, right, you need to use search instead of match.

You probably find it confusing because you think in terms of allowed characters and not of forbidden characters. But you actually specify a list of forbidden characters in your regular expression and apply a double negation, which is actually arguably harder to follow for the reader.

Reply:
I've edited my suggestion, it should work now.

Reply:
If I had to caricature, my suggestions says "is this string containing one of those forbidden characters", while your code says "is each character not a non-allowed character", and makes the "non empty" requirement implicit.

Reply:
if we go this way, i found it useless to keep regex, checking with in seems more clear.

Reply:
I prefer my version, but I'm fine with merging yours.

Reply:
I adapt you code without regex.

Reply:
Sorry, your comments just showed up in both my email inbox and on the github page. I saw your new version and its better. My points still remain and I don't see a reason for having both DOMAIN_REGEX and USER_REGEX if there are doing the same thing.

But as I said, I'm okay with merging, it is still very easy to read.

Reply:
These comments appearing late got to be a bit annoying. I'm okay with your last changes.